app/src/main/java/ml/docilealligator/infinityforreddit/adapters/CommentsRecyclerViewAdapter.java

```diff
@@ -736,44 +736,28 @@ public void onFetchMoreCommentSuccess(ArrayList<Comment> topLevelComments,
-
-                                                if (placeholderPosition != -1) {
-                                                    mVisibleComments.get(placeholderPosition).setLoadingMoreChildren(false);
-                                                    mVisibleComments.get(placeholderPosition).setLoadMoreChildrenFailed(true);
-                                                }
-                                                ((LoadMoreChildCommentsViewHolder) holder).placeholderTextView.setText(R.string.comment_load_more_comments_failed);
-                                            }
-
-                                            mVisibleComments.get(parentPosition).getChildren().get(mVisibleComments.get(parentPosition).getChildren().size() - 1)
-                                                    .setLoadingMoreChildren(false);
-                                            mVisibleComments.get(parentPosition).getChildren().get(mVisibleComments.get(parentPosition).getChildren().size() - 1)
-                                                    .setLoadMoreChildrenFailed(true);
-                                        } else {
-                                            for (int i = 0; i < mVisibleComments.size(); i++) {
-                                                if (mVisibleComments.get(i).getFullName().equals(parentComment.getFullName())) {
-                                                    if (mVisibleComments.get(i).isExpanded()) {
-                                                        int placeholderPositionHint = i + mVisibleComments.get(i).getChildren().size();
-                                                        int placeholderPosition = findLoadMoreCommentsPlaceholderPosition(parentComment.getFullName(), placeholderPositionHint);
-
-                                                        if (placeholderPosition != -1) {
-                                                            mVisibleComments.get(placeholderPosition).setLoadingMoreChildren(false);
-                                                            mVisibleComments.get(placeholderPosition).setLoadMoreChildrenFailed(true);
-                                                        }
-                                                        ((LoadMoreChildCommentsViewHolder) holder).placeholderTextView.setText(R.string.comment_load_more_comments_failed);
-                                                    }
+                                        int currentParentPosition = findCommentPosition(parentComment.getFullName(), parentPosition);
+                                        if (currentParentPosition == -1) {
+                                            // note: returning here is probably a mistake, because
+                                            // parent is just not visible, but it can still exist in the comments tree.
+                                            return;
```

Comment:
Just to be clear, this is the same behavior as before the refactoring. It's just more obvious now